src/pydicom/pixels/utils.py

```diff
@@ -354,9 +354,9 @@ def compress(
         will be added if needed for large amounts of compressed *Pixel
         Data*, otherwise just the basic offset table will be used.
     generate_instance_uid : bool, optional
-        If ``True`` (default) then generate a new (0008,0018) *SOP Instance UID*
-        value for the dataset using :func:`~pydicom.uid.generate_uid`, otherwise
-        keep the original value.
+        If ``True`` (default) and a lossy compression method is being used then
```

Comment:
I find it a bit contra-intuitive that you can set this to `True`, and no UID is generated anyway. I would probably let this default to `None`, and if `None`, set it to `True` for compressed and `False` for uncompressed syntax. If the user explicitely sets this, I would just take that setting. The disadvantage would be that the user will be able to create non-conformant DICOM by setting this to `False` for compressed data - but then this would be a conscious decision.
Just a thought...

Reply:
Changed